dvc/remote/gdrive.py

```diff
@@ -324,10 +325,40 @@ def _create_remote_dir(self, parent_id, title):
 
     @gdrive_retry
     def _delete_remote_file(self, remote_id):
+        from pydrive2.files import ApiRequestError
+
         param = {"id": remote_id}
         # it does not create a file on the remote
         item = self.drive.CreateFile(param)
-        item.Delete()
+
+        try:
+            item.Trash() if self._trash_only else item.Delete()
+        except ApiRequestError as exc:
+            if (
+                not exc.error.get("code", None)
+                and self.list_params["corpora"] != "drive"
+            ):
+                raise
+            if (
+                exc.error.get("code", None) == 403
+                and len(exc.error.get("errors", []))
+                and exc.error["errors"][0].get("location", "")
```

Comment:
do you know if there is a case of multiple errors?

Reply:
can't say for sure, haven't seen more than one. But just discovered that `content` on which we do `json.loads` might be `list` sometimes according to https://github.com/googleapis/google-api-python-client/blob/master/googleapiclient/errors.py#L58